gio/src/socket_control_message.rs

```diff
@@ -0,0 +1,39 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::prelude::*;
+use crate::SocketControlMessage;
+use glib::object::IsA;
+use glib::translate::*;
+use std::mem::MaybeUninit;
+
+impl SocketControlMessage {
+    #[doc(alias = "g_socket_control_message_deserialize")]
+    pub fn deserialize(level: i32, type_: i32, data: &[u8]) -> SocketControlMessage {
+        let size = data.len() as _;
+        unsafe {
+            from_glib_full(ffi::g_socket_control_message_deserialize(
+                level,
+                type_,
+                size,
+                data.to_glib_none().0 as *mut _,
+            ))
+        }
+    }
+}
+pub trait SocketControlMessageExtManual: Sized {
+    #[doc(alias = "g_socket_control_message_serialize")]
+    fn serialize(&self, data: &mut [MaybeUninit<u8>]);
+}
+
+impl<O: IsA<SocketControlMessage>> SocketControlMessageExtManual for O {
+    #[doc(alias = "g_socket_control_message_serialize")]
+    fn serialize(&self, data: &mut [MaybeUninit<u8>]) {
```

Comment:
We don't do that anywhere else. Maybe better `&mut [u8]` for consistency for now? Dealing with `MaybeUninit` slices is not very convenient at least